.github/workflows/test.yml

```diff
@@ -72,3 +72,44 @@ jobs:
           cd py4j-python
           # Java TLS tests are disabled until they can be fixed (refs #441)
           nosetests -v --exclude='.*java_tls_test.*'
+
+  test-doc:
+    name: Documentation build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout guidelines repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Install Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
+
+      - name: Install Python packages
+        run: |
+          python3 -m pip install -r py4j-web/requirements-doc.txt
+          python3 -m pip list
+
+      - name: Test Sphinx build
```

Comment:
This specific step can only be tested after this PR gets merged because git push origin gh-pages --force step can only be executed in the workflows running at apache/spark (it fails in forked repositories).

BTW, most of the logic below was copied from another project I maintain so it should work.